### ISSUE TITLE
Link to tslib on the packages page is broken

### ISSUE THREAD

**Issue by ygale** [haskell/hackage-server#800]
There is a package called "tslib" listed on the packages page, ostensibly the "front page" of hackage, very visible right near the top of the page. This is a broken link - `/package/tslib` is a 404.

See also #524. OK, I get it, we currently have no way to get rid of tslib from the index. But can we at least mark it somehow so that it doesn't show up as an ugly splotch on the site? It could even be hard-coded CSS or Javascript.

**Comment by hvr** [comment]
This turned out to be a quick with the Nginx configuration; you don't actually get a 404 but instead you get a 451 but with the wrong content-type header... and for some reason browsers make it look like a 404 to you:

```
$ curl -v http://hackage.haskell.org/package/tslib
> GET /package/tslib HTTP/1.1
> Host: hackage.haskell.org
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 451 
< Server: nginx/1.14.0 (Ubuntu)
< Date: Thu, 03 Jan 2019 11:24:45 GMT
< Content-Type: application/octet-stream
< Content-Length: 30
< Connection: keep-alive
< 
Unavailable For Legal Reasons
```

I've fixed this up, and now this should interact beter with browsers: http://hackage-origin.haskell.org/package/tslib

**Comment by ygale** [comment]
Hmm. Well if that's what the link is supposed to do, then so be it. 

But from a user perspective, is there a reason we want that link to appear on the Hackage page? Is it useful information to anyone?

Seems to me that since this happens so rarely, we could just manually remove it from the HTML, or put HTML/CSS/JS on the site so that this particular entry isn't visible, or whatever. And then do it again if this happens for some other package in the future.

**Comment by gbaz** [comment]
The proper "front page" is the browse page now, and since tslib is deprecated, its it goes all the way to the end: http://hackage.haskell.org/packages/browse

But yeah, in general we could do a better job "really" hiding tombstoned packages.

**Comment by ygale** [comment]
What do you mean that tslib "goes all the way to the end"? If I jump to the last page, it has packages starting with "z", tslib is not there.

That's the "proper front page"? I never saw it until now. The name "browse" is a bad name for this page - it's nearly useless for browsing. You can't display more than 100 packages at a time, and besides a jump to one of the first or last 5 pages, you can't move more than one page at a time. So to "browse" to the middle of the list would take hours. It's OK for searching though. I'll stick with the old packages page until this is improved.

**Comment by gbaz** [comment]
Fair enough on the difficulty of jumping to the middle. The lib we use for table display is pretty configurable, so PRs to improve that welcome. The in-page filter is quite nice though, among other things. 

I could have _sworn_ we had logic to push things to the end, when they were deprecated, but I'm seeing that we currently don't, whoops. We should just add filter-deprecated-by-default as part of improving the datatables stuff generally: https://github.com/haskell/hackage-server/issues/680
